### ISSUE TITLE
XMI Deserialization and serialization mismatch

### ISSUE THREAD

**Issue by arrys** [pyecore/pyecore#142]
Hi,

I've run into an issue when trying to deserialize an XMI model and then serialize it back to the original XMI.
The model differs from the original after being written.

I've made [a POC demonstrating my issue](https://github.com/arrys/pyecore-serialisation/tree/pre-generated).

To summarize the POC.
We have an ecore meta-model like this:

    <?xml version="1.0" encoding="UTF-8"?>
    <ecore:EPackage xmi:version="2.0" xmlns:xmi="http://www.omg.org/XMI" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:ecore="http://www.eclipse.org/emf/2002/Ecore" name="EXAMPLE" nsURI="http://www.example.org/EXAMPLE" nsPrefix="EXAMPLE">
      <eClassifiers xsi:type="ecore:EClass" name="Example">
        <eStructuralFeatures xsi:type="ecore:EAttribute" name="values" unique="false" lowerBound="1" upperBound="-1" eType="ecore:EDataType http://www.eclipse.org/emf/2002/Ecore#//EDouble" defaultValueLiteral="0.0"/>
      </eClassifiers>
    </ecore:EPackage>

An XMI file is read and then without making any changes written.
The written file has the values attribute instead of values children.
Reading in the file which has the attribute will yield a file with the attribute but with an empty list.

The original file looks like this:

    <?xml version="1.0" encoding="UTF-8"?>
    <EXAMPLE:Example xmi:version="2.0" xmlns:xmi="http://www.omg.org/XMI" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:EXAMPLE="http://www.example.org/EXAMPLE">
        <values>1.0</values>
        <values>2.0</values>
        <values>3.0</values>
    </EXAMPLE:Example>

After the first step:

    <?xml version='1.0' encoding='UTF-8'?>
    <EXAMPLE:Example xmlns:xmi="http://www.omg.org/XMI" xmlns:EXAMPLE="http://www.example.org/EXAMPLE" values="1.0 2.0 3.0" xmi:version="2.0"/>

After the second step:

    <?xml version='1.0' encoding='UTF-8'?>
    <EXAMPLE:Example xmlns:xmi="http://www.omg.org/XMI" xmlns:EXAMPLE="http://www.example.org/EXAMPLE" values="[]" xmi:version="2.0"/>

At this point a third step will fail on the deserialization part.

**Comment by aranega** [comment]
Hi @arrys, thanks a lot for the issue and really sorry about the late answer. 

I'll look into your issue very soon. Thanks a lot for the test case, it will help a lot to go faster to solve this :)

**Comment by aranega** [comment]
@arrys I fixed the bug on `develop`. Long story short, it's a regression I introduced few month back, when I considered a generator to deal with multivalues EAttributes.

The fix will always produce the good result, i.e: all the values will be inside the collection, but the values will be put in the same "line". Instead of having various entries for each values, you'll have something like the first temporary save that you have (the one with `values="1.0 2.0 3.0"`). This is not a problem as it's standard XMI/XML.
